joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -101,7 +101,9 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
       val baseStmtBlockAst = astForMethodBody(node.body, optionalStatementList)
       transformAsClosureBody(refs, baseStmtBlockAst)
     } else {
-      if (methodName != Defines.Initialize && methodName != Defines.InitializeClass) {
+      if (
+        methodName != Defines.Initialize && methodName != Defines.InitializeClass && methodName != Defines.TypeDeclBody
```

Comment:
Initialize should be the only one going to `astForConstructorMethodBody`
```suggestion
        methodName != Defines.Initialize
```

Reply:
`Defines.InitializeClass` can be completely removed too